pymc_marketing/mmm/tvp.py

```diff
@@ -0,0 +1,67 @@
+from typing import Optional
+
+import pymc as pm
+
+from pymc_marketing.mmm.utils import softplus
+
+
+def time_varying_prior(
+    name: str,
+    X: pm.Deterministic,
+    X_mid: int | float,
+    dims: Optional[tuple[str, str] | str] = None,
+    m: int = 40,
+    L: int | float = 100,
+    eta_lam: float = 1,
+    ls_mu: float = 5,
+    ls_sigma: float = 5,
+    cov_func: Optional[pm.gp.cov.Covariance] = None,
+) -> pm.Deterministic:
+    """Time varying prior, based the Hilbert Space Gaussian Process (HSGP).
```

Comment:
Can you add a reference to the pymc classes and some relevant papers (you can take them from https://www.pymc.io/projects/docs/en/stable/api/gp/generated/pymc.gp.HSGP.html)

Reply:
✅